File: 2015/07/2.py
from parse import parse
from calculate import calculate

input = None
try:
    input = open("./2015/07/input.txt", "r")
except:
    print("Error opening input.txt")
    exit(1)
parsed = parse(input.read())
a = calculate(parsed, "a")

for key in parsed:
    parsed[key]["status"] = "NOT_CALCULATED"

parsed["b"]["status"] = "CALCULATED"
parsed["b"]["calculated"] = a

print(calculate(parsed, "a"))

File: 2015/07/2.js
"use strict";
const fs = require("node:fs");
const parse = require("./parse");
const calculateSignal = require("./calculateSignal");
let contents = fs.readFileSync("./2015/07/input.txt", "utf8").toString();

let instructions = parse(contents);
const a = calculateSignal(instructions, "a");

for (let key in instructions) {
  instructions[key].state = undefined;
}

instructions["b"] = {
  state: "CALCULATED",
  calculatedValue: a,
};

console.log(calculateSignal(instructions, "a"));
